modules/core/src/test/java/org/apache/ignite/internal/util/DisabledStripedExecutorTest.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.ignite.spi.discovery.tcp.ipfinder.TcpDiscoveryIpFinder;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+/**
+ * Striped thread pool disabling test
+ */
+public class DisabledStripedExecutorTest extends GridCommonAbstractTest {
+    /** */
+    private static final TcpDiscoveryIpFinder IP_FINDER = new TcpDiscoveryVmIpFinder(true);
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String gridName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(gridName);
+
+        cfg.setDiscoverySpi(new TcpDiscoverySpi().setIpFinder(IP_FINDER));
+
+        return cfg;
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    public void testStripedExecutorService() throws Exception {
+        checkStripedExecutorService(-1);
+
+        checkStripedExecutorService(0);
+
+        checkStripedExecutorService(2);
+    }
+
+    /**
+     * Check that striped pool is disabled if size is not a positive value.
+     *
+     * @param stripedPoolSize Striped pool size.
+     * @throws Exception If failed.
+     */
+    private void checkStripedExecutorService(int stripedPoolSize) throws Exception {
+        IgniteConfiguration cfg = getConfiguration();
+
+        cfg.setStripedPoolSize(stripedPoolSize);
+
+        try (Ignite ignite = startGrid("grid" + stripedPoolSize, cfg)) {
+            final GridKernalContext ctx = ((IgniteKernal)ignite).context();
+
+            if (stripedPoolSize < 1) {
+                assertTrue("Striped Executor should be disabled",
+                    ctx.isStripedExecutorDisabled());
+
+                // getStripedExecutorService should throw AssertionError.
+                GridTestUtils.assertThrows(log, new Callable<Object>() {
+                    @Override public Object call() {
+                        ctx.getStripedExecutorService();
+
+                        return null;
+                    }
+                }, AssertionError.class, null);
+            }
+            else {
+                assertFalse("Striped Executor should be enabled",
+                    ctx.isStripedExecutorDisabled());
+
+                // getStripedExecutorService should not throw AssertionError.
+                ctx.getStripedExecutorService();
+            }
+
+            checkCacheOperations(ignite);
+        }
+    }
+
+    /**
+     * Check that GET and PUT cache operations work correctly.
+     *
+     * @param ignite Ignite instance.
+     */
+    private void checkCacheOperations(Ignite ignite) {
+        IgniteCache<Integer, Integer> cache = ignite.getOrCreateCache("temp-cache");
+
+        Integer key = 1;
+        Integer val = 1;
+
+        cache.put(key, val);
+
+        assertEquals("Cache should contain saved object", val, cache.get(key));
+    }
+}
```

Comment:
Need to add balnk line - file should finish with blank one

Reply:
Fixed